src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -3714,6 +3708,17 @@ void SmartScript::SortEvents(SmartAIEventList& events)
     std::sort(events.begin(), events.end());
 }
 
+void SmartScript::RaisePriority(SmartScriptHolder& e)
+{
+    e.timer = 1;
+    // Change priority only if it's set to default, otherwise keep the current order of events
+    if (e.priority == SmartScriptHolder::DEFAULT_PRIORITY)
+    {
+        e.priority = mCurrentPriority++;
```

Comment:
```suggestion
        e.priority = ++mCurrentPriority;
```

Reply:
that changes the way the code works though